sdk/attestation/attestation/src/models/attestationToken.ts

```diff
@@ -33,12 +33,31 @@ import { base64EncodeByteArray } from "../utils/base64";
  *  provided, they are all assumed to be 'true'.
  *
  */
-export interface TokenValidationOptions {
+export interface AttestationTokenValidationOptions {
+  /**
+   * If true, validate the attestation token, if false, skip validation.
+   */
+  validateToken?: boolean;
+  /**
+   * The expected issuer for the {@link AttestationToken}. Only checked if {@link validateIssuer} is set.
+   */
   expectedIssuer?: string;
   validateExpirationTime?: boolean;
   validateNotBeforeTime?: boolean;
-  validateToken?: boolean;
+  validateIssuer?: boolean;
   timeValidationSlack?: number;
+
+  /**
+   * Validation Callback which allows customers to provide their own validation
+   * functionality for the attestation token. This can be used to validate
+   * the signing certificate in AttestationSigner.
+   *
+   * @remarks
+   *
+   * If there is a problem with token validation, the validaitonCallback is expected
+   * to throw an exception.
+   */
+  validationCallback?: (token: AttestationToken, signer: AttestationSigner | null) => void;
```

Comment:
why null and not optional?

Reply:
For some reason, that caused an error when I first tried it. Trying again.